src/test_hash.py

```diff
@@ -0,0 +1,216 @@
+# -*- coding: utf-8 -*-
+
+import pytest
+import io
+import sys
+import random
+import os
+
+dir_path = os.path.dirname(os.path.realpath(__file__))
+file = io.open(dir_path + '/subset.txt')
+words = file.read()
+file.close()
```

Comment:
A safer way to handle opening files is to use the context handler.

Instead of open, read, close

```with open(file, options) as file:
    words = file.read()```

If something goes wrong with the file read, the first version might not close the file -- resulting in some problems. The context handler ensures that the file is always closed, no matter what.